src/encode.rs

```diff
@@ -113,30 +113,102 @@ pub fn encode_to_vec(data: impl AsRef<[u8]>, width: u32, height: u32) -> Result<
     Encoder::new(&data, width, height)?.encode_to_vec()
 }
 
+pub struct EncoderBuilder<'a> {
+    data: &'a [u8],
+    width: u32,
+    height: u32,
+    stride: Option<usize>,
+    raw_channels: Option<RawChannels>,
+    colorspace: Option<ColorSpace>,
+}
+
+impl<'a> EncoderBuilder<'a> {
+    /// Creates a new encoder builder from a given array of pixel data and image dimensions.
+    pub fn new(data: &'a (impl AsRef<[u8]> + ?Sized), width: u32, height: u32) -> Self {
+        Self {
+            data: data.as_ref(),
+            width,
+            height,
+            stride: None,
+            raw_channels: None,
+            colorspace: None,
+        }
+    }
+
+    /// Set the stride of the pixel data.
+    pub const fn stride(mut self, stride: usize) -> Self {
+        self.stride = Some(stride);
+        self
+    }
+
+    /// Set the input format of the pixel data.
+    pub const fn raw_channels(mut self, raw_channels: RawChannels) -> Self {
+        self.raw_channels = Some(raw_channels);
+        self
+    }
```

Comment:
Suggestion: raw->source. "Raw" is very non-self-descriptive imo
```suggestion
    /// Set the source format of the pixel data.
    pub const fn source_channels(mut self, source_channels: SourceChannels) -> Self {
        self.source_channels = Some(source_channels);
        self
    }
```

Reply:
ack